src/analyses/interval_domain.h

```diff
@@ -56,26 +56,40 @@ class interval_domaint:public ai_domain_baset
   }
 
   // no states
-  void make_bottom() final
+  void make_bottom() final override
   {
     int_map.clear();
     float_map.clear();
     bottom=true;
   }
 
   // all states
-  void make_top() final
+  void make_top() final override
   {
     int_map.clear();
     float_map.clear();
     bottom=false;
   }
 
-  void make_entry() final
+  void make_entry() final override
   {
     make_top();
   }
 
+  bool is_bottom() const override final
+  {
+    // This invariant should hold but is not correct enforced at the moment.
```

Comment:
Nit: `correct`->`correctly`. 

If it is known, explaining _when_ it is not correctly enforced would greatly improve this comment.

Reply:
Fixed.

I don't know for sure.  My guess is that merge or transform for assume/goto is setting it to bottom in some way that isn't expected / correct.  I know Lucas was chasing some issues like that.  The patch that adds this originally came from @danpoe and he had added it commented out.  Given that there are design issues with this domain (like, it doesn't handle pointers, at all) and we are writing a replacement (which will) I'm not inclined to spend too much time on this.  So I'm content for it to be documented as a design aim but not enforced.